core/include/gnuradio-4.0/Message.hpp

```diff
@@ -20,11 +20,41 @@ const std::string ErrorInfo = "ERROR_INFO_KEY"; // optional: if a message has an
 
 namespace message::kind {
 const std::string Error           = "ERROR_KIND";
-const std::string Graph_update    = "GRAPH_UPDATE_KIND";
+const std::string GraphUpdate     = "GRAPH_UPDATE_KIND";
+const std::string SchedulerUpdate = "SCHEDULER_UPDATE_KIND";
 const std::string UpdateSettings  = "UPDATE_SETTINGS_KIND";
 const std::string SettingsChanged = "SETTINGS_CHANGED_KIND";
 } // namespace message::kind
 
+namespace message::graph_update {
+namespace what {
+const std::string BlockAdded         = "ADDED_BLOCK_GRAPH_UPDATE";
+const std::string BlockRemoved       = "REMOVED_BLOCK_GRAPH_UPDATE";
+const std::string BlockExchanged     = "EXCHANGED_BLOCK_GRAPH_UPDATE";
+const std::string BlocksConnected    = "CONNECTED_BLOCKS_GRAPH_UPDATE";
+const std::string BlocksDisconnected = "DISCONNECTED_BLOCKS_GRAPH_UPDATE";
```

Comment:
There's the initial creation of the UI elements, which could be done by the scheduler sending an array of BlockAdded/BlocksConnected messages, or a single GraphDescription message (that might require a quite complex pmt structure, though). Alternatively, the UI could read the structure from the graph before passing it to the scheduler. I'd prefer the single channel (messages) though.

Reply:
Would BlockAdded/BlockRemoved/ConnectionAdded/BlocksConnected/BlocksDisconnected be accompanied by AddBlock/RemoveBlock/ConnectBlocks/DisconnectBlocks commands, to trigger these from the UI? Or should it be the same command code for both directions?